### PR TITLE
Convert MapSnapshotOptions to struct

### PR DESCRIPTION
**Pull request by julianrex**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [X] Briefly describe the changes in this PR.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [x] Document any changes to public APIs.
 - [X] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

This PR converts `MapSnapshotOptions` to a struct, inline with other value types.

### CODE REVIEW

File: Sources/MapboxMaps/Foundation/Extensions/Core/MapSnapshotOptions.swift [hunk 1]
```diff
@@ -11,24 +23,38 @@ extension MapSnapshotOptions {
...
+                pixelRatio: CGFloat,
+                glyphsRasterizationOptions: GlyphsRasterizationOptions? = GlyphsRasterizationOptions(),
+                resourceOptions: ResourceOptions = ResourceOptionsManager.default.resourceOptions) {
         precondition(pixelRatio > 0)
         precondition(size.width * pixelRatio <= 8192, "Width or scale too great.")
         precondition(size.height * pixelRatio <= 8192, "Height or scale too great.")
 
-        let coreOptions = MapboxCoreMaps.ResourceOptions(resourceOptions)
+        self.size = size
+        self.pixelRatio = pixelRatio
```
Comment (neelmistry94) [inline Sources/MapboxMaps/Foundation/Extensions/Core/MapSnapshotOptions.swift:35]: Do we need to do any magic here with pixel ratio with snapshotter like we had to do for the car play scenario?

Comment (julianrex) [reply Sources/MapboxMaps/Foundation/Extensions/Core/MapSnapshotOptions.swift:35]: We should be good here.

File: Sources/MapboxMaps/Foundation/Extensions/Core/MapSnapshotOptions.swift [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+public struct MapSnapshotOptions {
+    /// Dimensions of the snapshot in points.
+    public let size: CGSize
+
+    /// Ratio between the number device-independent and screen pixels.
+    public let pixelRatio: CGFloat
+
+    /// Glyphs rasterization options to use for client-side text rendering.
+    /// By default, `GlyphsRasterizationOptions` will use `.ideographsRasterizedLocally`
+    public let glyphsRasterizationOptions: GlyphsRasterizationOptions?
```
Comment (nishant-karajgikar) [inline Sources/MapboxMaps/Foundation/Extensions/Core/MapSnapshotOptions.swift:11]: Could we set the default here?

**Review by nishant-karajgikar — APPROVED** [review]

File: Sources/MapboxMaps/Foundation/Extensions/Core/MapSnapshotOptions.swift (hunk 2 above)
Comment (julianrex) [reply Sources/MapboxMaps/Foundation/Extensions/Core/MapSnapshotOptions.swift:11]: Yep; we might want to keep the type as optional though matching the underlying type. Adding a default would match what we had prior to this PR.

**Pull request merged**